src/frontends/tensorflow_common/src/op/binary_op.cpp

```diff
@@ -169,6 +169,47 @@ OutputVector translate_addv2_op(const NodeContext& node) {
     return {result};
 }
 
+OutputVector translate_equal_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"Equal"}, true);
+    auto x = node.get_input(0);
+    auto y = node.get_input(1);
+
+    auto complex_type_mark_x = as_type_ptr<ComplexTypeMark>(x.get_node_shared_ptr());
+    auto complex_type_mark_y = as_type_ptr<ComplexTypeMark>(y.get_node_shared_ptr());
+
+    // If both inputs are complex tensors
+    if (complex_type_mark_x && complex_type_mark_y) {
+        element::Type complex_part_type_x = complex_type_mark_x->get_complex_part_type();
+        element::Type complex_part_type_y = complex_type_mark_y->get_complex_part_type();
+
+        auto tensor1 = complex_type_mark_x->input_value(0);
+        auto tensor2 = complex_type_mark_y->input_value(0);
+
+        auto equal_op = make_shared<v1::Equal>(tensor1, tensor2);
```

Comment:
check output shape. What shape will be in TF?

Reply:
same problem with output shape. What should be done additionally right after equal? Check `Reduce*` ops

Reply:
@MonalSD, please think why do you need `ReduceAnd` (check OV opset) operation for this implementation.

Reply:
> Reduce
@rkazants I'm not able to understand how reduceAnd operation will be implemented for the equal operation, like are we trying to get the final result as a single element tensor?

Reply:
Hi @MonalSD, please think why `ReduceAnd` should be used. Equal is element-wise operation and complex numbers are equal when both real and imag parts are equal. So you should reduce boolean tensor by last dimension using `ReduceAnd`.